modules/tensor_mechanics/src/kernels/StressDivergenceRZTensors.C

```diff
@@ -0,0 +1,130 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "StressDivergenceRZTensors.h"
+
+template<>
+InputParameters validParams<StressDivergenceRZTensors>()
+{
+  InputParameters params = validParams<StressDivergenceTensors>();
+  params.addClassDescription ("Calculate stress divergence for an axisymmetric problem in cylinderical coordinates.");
+  params.addRequiredParam<unsigned int>("component", "An integer corresponding to the direction the variable this kernel acts in. (0 for x, 1 for y, 2 for z; note in this kernel disp_x refers to the radial displacement and disp_y refers to the axial displacement.)");
+  params.set<bool>("use_displaced_mesh") = true;
+  return params;
+}
+
+StressDivergenceRZTensors::StressDivergenceRZTensors(const std::string & name, InputParameters parameters) :
+    StressDivergenceTensors(name, parameters)
+{
+}
+
+Real
+StressDivergenceRZTensors::computeQpResidual()
+{
+  Real div = 0;
+
+  if (_assembly.coordSystem() != Moose::COORD_RZ)
+    mooseError("The coordinate system in the Problem block must be set to RZ for Axisymmetric geometries.");
+
+  if (_component == 0)
+  {
+    div = _grad_test[_i][_qp](0) * _stress[_qp](0,0) +
+        + ( _test[_i][_qp] / _q_point[_qp](0)) * _stress[_qp](2,2) +
+        + _grad_test[_i][_qp](1) * _stress[_qp](0,1); // stress_{rz}
+//      + 0 * _stress[_qp](1,1)
```

Comment:
The commented out stress in line 37 will never be called; I left the comment to show that the stress residual for stress_xx does not depend on stress_yy.  Jason used this comment style in solid mechanics, and, since I found the comments helpful to understanding the cylindrical coordinates divergence, I left the comment in here as well.  I'll remove that line to conform to the code standard (pending your response to my question below).